src/ValueHandler/ProductOptionValueHandler.php

```diff
@@ -161,16 +211,130 @@ public function handle($productVariant, string $optionCode, array $akeneoValue):
         }
     }
 
-    private function isVariantOption(ProductVariantInterface $productVariant, string $attribute): bool
+    private function handleMetricOption(ProductOptionInterface $productOption, string $optionCode, array $akeneoDataValue, ProductInterface $product, ProductVariantInterface $productVariant): void
     {
-        $product = $productVariant->getProduct();
-        Assert::isInstanceOf($product, ProductInterface::class);
-        foreach ($product->getOptions() as $option) {
-            if ($attribute === $option->getCode()) {
-                return true;
+        if (!array_key_exists('amount', $akeneoDataValue)) {
+            throw new LogicException('Amount key not found');
+        }
+        $floatAmount = (string) ($akeneoDataValue['amount']);
+        if (!array_key_exists('unit', $akeneoDataValue)) {
+            throw new LogicException('Unit key not found');
+        }
+        $unit = (string) $akeneoDataValue['unit'];
+        $optionValueCode = $optionCode . '_' . $floatAmount . '_' . $unit;
```

Comment:
We shouldn't put the dot symbol in the code.
Generally, in Sylius all code values "can only be comprised of letters, numbers, dashes and underscores.".

Reply:
Right! Didn't think about it. What do you suggest to do?

Reply:
Maybe a sort of "slugification" of the string? We could extract a method that generates a clean optino value code.

Reply:
I think that it's ok to replace the dot with nothing. For example: if the amount returned from Akeneo is "250.0000" the code will be "option-code_2500000_ecc". If the amount returned from Akeneo is "2500000.0000" the code will be "option-code_25000000000_ecc". Shouldn't get any conflicts.

This is based on the fact that Akeneo will always have the same precision (of 4 in this case) for float values. If this is not true, we should force it by using number_format with a specific precision.